File: PasswordGenerator.cs
public class PasswordGenerator
{
    private const string Lowercase = "qwertyuiopasdfghjklzxcvbnm"; 
    private const string Digits = "0123456789";
    private const string Symbols = "!@#$%^&*()_+=-\"№;:?[]{}/\\.,<>`~";
    
    public string GeneratePassword(uint length, bool includeLowercase = true, bool includeUppercase = true, bool includeDigits = true, bool includeSymbols = true)
    {
        var allSymbols = "";
        allSymbols += includeLowercase switch
        {
            false when !includeUppercase => Lowercase,
            false when includeUppercase => Lowercase.ToUpper(),
            _ => Lowercase.ToLower()
        };
        
        allSymbols = includeDigits ? allSymbols + Digits : allSymbols;
        
        return GeneratePassword(length, allSymbols);
    }

    private string GeneratePassword(uint length, string allSymbols)
    {
        var password = "";
        var random = new Random();
        for (uint i = 0; i < length; i++) password += allSymbols[random.Next(0, allSymbols.Length)];
        
        return password;
    }
}

File: password_generator.py
from random import randint

lowercase = "qwertyuiopasdfghjklzxcvbnm"
digits = "0123456789"
symbols = "!@#$%^&*()_+=-\"№;:?[]{}/\\.,<>`~"


def gen(all_symbols: str, length: int = 10):
    password = ''

    for i in range(length):
        password += all_symbols[randint(0, len(all_symbols) - 1)]
    return password

def generate_password(
        length: int,
        include_lowercase: bool = True,
        include_uppercase: bool = True,
        include_digits: bool = True,
        include_symbols:bool = True):
    all_symbols: str = ""

    if include_symbols == False and include_uppercase == False and include_digits == False and include_lowercase == False:
       all_symbols = lowercase
    else:
        all_symbols += lowercase if include_lowercase == True else ''
        all_symbols += lowercase.upper() if include_uppercase == True else ''
        all_symbols += digits if include_digits == True else ''
        all_symbols += symbols if include_symbols == True else ''

    return gen(all_symbols, length)
